thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/DashboardResource.java

```diff
@@ -438,6 +458,88 @@ public String getContributorData(@QueryParam("dataset") String collection,
     return jsonResponse;
   }
 
+  /**
+   * Get the timeseries with baseline for an anomaly function
+   * @param functionId
+   *      the target anomaly function
+   * @param startTimeIso
+   *      the start time
+   * @param endTimeIso
+   * @return
+   * @throws Exception
+   */
+  @GET
+  @Path(value = "/timeseries/{id}")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Map<String, AnomalyTimelinesView> getTimeSeriesAndBaselineData(@PathParam("id") long functionId,
+      @QueryParam("start") String startTimeIso, @QueryParam("end") String endTimeIso) throws Exception {
+    AnomalyFunctionDTO anomalyFunctionSpec = DAO_REGISTRY.getAnomalyFunctionDAO().findById(functionId);
+    if(anomalyFunctionSpec == null) {
+      LOG.warn("Cannot find anomaly function {}", functionId);
+      return null;
+    }
+
+    DateTime startTime = new DateTime(0); // Set start time = 0 in default, so that all the baseline is exposed
+    DateTime endTime = DateTime.now();
+    if (StringUtils.isNotBlank(startTimeIso)) {
+      startTime = ISODateTimeFormat.dateTimeParser().parseDateTime(startTimeIso);
+    }
+    if (StringUtils.isNotBlank(endTimeIso)) {
+      endTime = ISODateTimeFormat.dateTimeParser().parseDateTime(endTimeIso);
+    }
+
+    if (startTime.isAfter(endTime)) {
+      LOG.warn("Start time {} is after end time {}", startTime, endTime);
+      return null;
+    }
+
+    Map<String, AnomalyTimelinesView> dimensionMapAnomalyTimelinesViewMap = new HashMap<>();
+    AnomalyDetectionInputContextBuilder anomalyDetectionInputContextBuilder =
+        new AnomalyDetectionInputContextBuilder(anomalyFunctionFactory);
+    anomalyDetectionInputContextBuilder.setFunction(anomalyFunctionSpec);
+    // Take start time == end time to generate the baseline before end time
+    AnomalyDetectionInputContext anomalyDetectionInputContext = anomalyDetectionInputContextBuilder
+        .fetchTimeSeriesData(endTime, endTime, true).fetchExistingMergedAnomalies(endTime, endTime, false)
```

Comment:
@cyenjung  yes. I should pass the start time equals 1 bucket before the end time.
@kishoreg That function is based on an anomaly result in the system. I will see if I can reuse the code.